uk.co.compucorp.civicrm.civihrsampledata/CRM/CiviHRSampleData/Importers/AbsencePeriod.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+
+/**
+ * Class CRM_CiviHRSampleData_Importers_AbsencePeriod
+ *
+ */
+class CRM_CiviHRSampleData_Importers_AbsencePeriod extends CRM_CiviHRSampleData_DataImporter
+{
+
+  public function __construct() {
+    $this->removeAllPeriods();
+  }
+
+  /**
+   * @see CRM_CiviHRSampleData_DataImporter::insertRecord
+   * @param array $row
+   */
+  protected function insertRecord(array $row) {
+    $this->callAPI('HRAbsencePeriod', 'create', $row);
+  }
+
+  /**
+   * Remove any existing absence period.
+   */
+  private function removeAllPeriods() {
+    $query = "DELETE FROM civicrm_hrabsence_period";
```

Comment:
the API for delete only work for single record but I need to remove any defined absence period .. also I think it is more efficient  than fetching all absence periods via the API and then loop through the results and delete them one by one  . ( but anyway there is probably only single absence period on fresh installations so I think that might work )